src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -3215,7 +3215,10 @@ void ApplicationManagerImpl::UnregisterApplication(
     MessageHelper::SendUnsubscribedWayPoints(*this);
   }
 
-  EndNaviServices(app_id);
+  if (navi_app_to_stop_.end() ==
+      std::find(navi_app_to_stop_.begin(), navi_app_to_stop_.end(), app_id)) {
+    EndNaviServices(app_id);
+  }
```

Comment:
@LitvinenkoIra Could you be so kind and explain the logic here, please

Reply:
@AByzhynar In `EndNaviServices` function we add the `app_id` of application that should complete the services to the `navi_app_to_stop_` collection.
 So here we just need to check - if `app_id` already presents in  `navi_app_to_stop_`  collection -  we don’t need to call EndNaviServices, because we already called it once